### PR TITLE
Migrate treemap.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/charts/treemap.dart [hunk 1]
```diff
@@ -18,28 +16,28 @@ enum PivotType { pivotByMiddle, pivotBySize }
...
-    @required this.levelsVisible,
+    required this.levelsVisible,
     this.isOutermostLevel = false,
-    @required this.width,
-    @required this.height,
-    @required this.onRootChangedCallback,
+    required this.width,
+    required this.height,
+    required this.onRootChangedCallback,
   });
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/charts/treemap.dart:26]: we should add an assert for both of these constructors that `rootNode == null != nodes == null`

File: packages/devtools_app/lib/src/charts/treemap.dart [hunk 2]
```diff
@@ -421,7 +419,7 @@ class _TreemapState extends State<Treemap> {
   /// ----------------------------
   /// ```
   Widget buildTreemap(BuildContext context) {
-    if (widget.rootNode.children.isNotEmpty) {
+    if (widget.rootNode!.children.isNotEmpty) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/charts/treemap.dart:422]: add an assert at the beginning of this method to verify widget.rootNode != null

File: packages/devtools_app/lib/src/charts/treemap.dart [hunk 3]
```diff
@@ -370,7 +368,7 @@ class _TreemapState extends State<Treemap> {
           return const SizedBox();
         }
         final positionedChildren = buildTreemaps(
-          children: widget.nodes,
+          children: widget.nodes!,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/charts/treemap.dart:371]: add an assert at the beginning of this method: `assert(widget.nodes?.isNotEmpty ?? false)`

File: packages/devtools_app/lib/src/charts/treemap.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/charts/treemap.dart:26]: done

File: packages/devtools_app/lib/src/charts/treemap.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/charts/treemap.dart:422]: done

File: packages/devtools_app/lib/src/charts/treemap.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/charts/treemap.dart:371]: `widget.nodes != null && widget.nodes!.isNotEmpty` seems to be more readable

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**